library/src/test/scala/SchemaExample.scala

```diff
@@ -632,6 +632,138 @@ object NewSchema {
           |    }
           |}""".stripMargin)
 
+  val completeExampleCodeSerializer =
+    """package com.example
+      |import _root_.sjsonnew._
+      |import _root_.sjsonnew.BasicJsonProtocol._
```

Comment:
BasicJsonProtocol is useful only for demo purpose.
I think what you should generate is a trait with a self type annotation of any necessary parent JSON protocols.